packages/flutter/lib/src/widgets/text_editing_intents.dart

```diff
@@ -0,0 +1,145 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'actions.dart';
+
+/// An [Intent] representing the default text editing behavior for the current
+/// platform for the alt + arrow down key event.
+class AltArrowDownTextIntent extends Intent {}
```

Comment:
The naming seems odd, don't we usually name intents after what they do, not after what key triggers them? e.g. ScrollIntent, ActivateIntent, and so on...

Reply:
I have a bit about this in the [design doc's open questions](https://docs.google.com/document/d/1QaVIr1bbOWJGNyyNsB75sXgTDRMuxP268sLGTBNDut0/edit#heading=h.dy33bukwh2dy).  I seem to be alone in preferring this kind of naming so I'm going to investigate changing it starting with the actions.

For the intents, my main point is that I'd like the visual text editing widgets to be ignorant of how text editing works.  This doesn't really affect these keyboard intents, but I'm also going to invoke intents from the visual text editing widgets like `Actions.invoke<SingleTapUpTextIntent>`.  If the intent is named for what it does and/or the widget decides which intent to invoke based on the platform or other things, then it's affecting business logic.

Reply:
If I change just the actions naming, this is the result: https://github.com/justinmc/flutter/blob/text-editing-refactor-action-names/packages/flutter/lib/src/widgets/text_editing_actions.dart

Reply:
A bunch of us met and decided on a hybrid approach.  The [open questions](https://docs.google.com/document/d/1QaVIr1bbOWJGNyyNsB75sXgTDRMuxP268sLGTBNDut0/edit#heading=h.dy33bukwh2dy) section has been updated with a detailed explanation.